### PR TITLE
Don't let organizers confirm competitions

### LINKED ISSUE #202: Only let delegates (not organizers!) confirm a competition
Asked for by @SAuroux [here](https://groups.google.com/d/msg/wca-software/uu33iVYJDQQ/i5u0HmcvDQAJ)

### PR DESCRIPTION
**Pull request by jfly**
This fixes #202 

@timhabermaas, could you take a look? Thanks!

### CODE REVIEW

File: WcaOnRails/app/views/competitions/edit.html.erb [hunk 1]
```diff
@@ -40,26 +41,26 @@
         </div>
       </div>
     <% else %>
-      <% if @competition.isConfirmed? && @competition.showAtAll? %>
+      <% if is_actually_confirmed && @competition.showAtAll? %>
```
Comment (timhabermaas) [inline WcaOnRails/app/views/competitions/edit.html.erb]: Does `@competition.isConfirmed?` not work?

File: WcaOnRails/spec/controllers/competitions_controller_spec.rb [hunk 2]
```diff
@@ -213,6 +213,12 @@
         sign_in organizer
       end
 
+      it 'cannot confirm competition' do
+        patch :update, id: competition, competition: { name: competition.name }, commit: "Confirm"
```
Comment (timhabermaas) [inline WcaOnRails/spec/controllers/competitions_controller_spec.rb]: A different route for confirming competitions would slightly clean up the controller and make this spec more "safe" (a typo in the `commit` value would probably lead to a passing test).

File: WcaOnRails/app/views/competitions/edit.html.erb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/views/competitions/edit.html.erb]: Does the commit message here: https://github.com/jfly/worldcubeassociation.org/commit/71454c814d6af91d6539fe3139f61288fc9fe8a6 make sense?

I was a bad boy and didn't write a test for this, under the argument that it's "just" ui code (something I haven't been very rigorous about testing). This is subtle enough that maybe it deserves a test. What do you think?

File: WcaOnRails/spec/controllers/competitions_controller_spec.rb (hunk 2 above)
Comment (jfly) [inline WcaOnRails/spec/controllers/competitions_controller_spec.rb]: That's a good point, this test is a little scary. However, a new route seems problematic, as we only have 1 form here. Are you envisioning writing some javascript client side to dispatch the form submission to the correct route?

Comment (timhabermaas) [inline WcaOnRails/spec/controllers/competitions_controller_spec.rb]: Depending on the layout you could either add another form just for the confirmation button or use the [`formaction`](http://devdocs.io/html/element/button) attribute to overwrite the `form[action]` URL. I've never used the latter option and am not sure about browser compatibility, though.

Comment (timhabermaas) [inline WcaOnRails/spec/controllers/competitions_controller_spec.rb]: Argh, looks like it requires IE10: http://www.w3schools.com/tags/att_input_formaction.asp

Adding JS is probably not worth it.

Comment (jfly) [inline WcaOnRails/spec/controllers/competitions_controller_spec.rb]: Oh, neat! I did not know about `formaction`. I'm inclined to leave this as is, however.

File: WcaOnRails/app/views/competitions/edit.html.erb (hunk 1 above)
Comment (timhabermaas) [inline WcaOnRails/app/views/competitions/edit.html.erb]: Oh, I get it now. I was under the impression that `@competition == Competition.find(@competition.id)`.

I wouldn't test it either if it's just some usability thing. But I think one simple feature test for editing a competition might be worth writing in order to catch stupid type errors.

Comment (jfly) [inline WcaOnRails/app/views/competitions/edit.html.erb]: I've added a simple feature test.

Comment (timhabermaas) [inline WcaOnRails/app/views/competitions/edit.html.erb]: :+1: feel free to merge it. The rest looks fine to me.

**Pull request merged**